backend/vkt/src/main/java/fi/oph/vkt/service/PublicReservationService.java

```diff
@@ -14,10 +15,10 @@ public class PublicReservationService {
 
   @Transactional
   public void deleteReservation(final long reservationId) {
-    try {
-      reservationRepository.deleteById(reservationId);
-    } catch (final Exception e) {
-      throw new NotFoundException(String.format("Reservation by id:%d not found", reservationId));
+    if (!reservationRepository.existsById(reservationId)) {
+      throw new NotFoundException(String.format("Reservation by id: %d not found", reservationId));
     }
+
+    reservationRepository.deleteAllByIdInBatch(List.of(reservationId));
```

Comment:
Tähän voisi ehkä lisätä kommenttina lyhyen selityksen syystä batch deleten käyttöön?

Reply:
Toinen vaihtoehto mikä itsellä kävi tätä toteuttaessa mielessä että kannattaiskohan luoda uusi abstrakti `BaseRepository` luokka, joka implementoisi metodin `deleteById`, jossa olisi tällainen toteutus eli "exists tarkistus" + "delete in batch". Laittaisi nykyiset repositoriot perimään sen JpaRepositoryn sijaan.

Reply:
Tää kuulostaa myös ihan järkevältä.

Reply:
Commitoin nyt BaseRepositoryn, jonka kaikki VKT:n repositoryt voisi implementoida jos vaikuttaa järkevältä. Kommentit tuosta metodista tosin unohtui ja sinne vois jotain lisätä. Tän PR:n vois ehkä sen jälkeen mergetä ja toteuttaa omina pull requesteinaan samat muutokset AKR:ään ja OTR:ään?

Reply:
Joo, tämä käy.